scripts/performance/main.py

```diff
@@ -0,0 +1,216 @@
+        if name in cat_data:
+            try:
+                cat_data[name].append(val)
+            except AttributeError:
+                cat_data[name] = [cat_data[name], val]
+        else:
+            cat_data[name] = val
+
+
+class DataRecorder(object):
+    def __init__(self, data):
+        self._data = data
+        self._timer = TicTocTimer()
+        self._category = {}
+
+        self._timingHandler = TimingHandler()
+        timing_logger = logging.getLogger('pyomo.common.timing')
+        timing_logger.setLevel(logging.INFO)
+        timing_logger.addHandler(self._timingHandler)
+
+    @unittest.pytest.fixture(autouse=True)
+    def add_data_attribute(self, request):
+        # set a class attribute on the invoking test context
+        request.cls.testdata = OrderedDict()
+        self._data[request.node.nodeid] = request.cls.testdata
+        yield
+        request.cls.testdata = None
+
+    @unittest.pytest.hookimpl(hookwrapper=True)
+    def pytest_runtest_call(self, item):
+        self._timingHandler.setTest(self._data[item.nodeid])
+        # Trigger garbage collection (try and get a "clean" environment)
+        gc.collect()
+        gc.collect()
+        self._timer.tic("")
+        # Run the test
+        yield
+        # Collect the timing and clean up
+        self._data[item.nodeid]['test_time'] = self._timer.toc("")
+        self._timingHandler.clearTest()
+
+
+def getPyomoInfo():
+    cwd = os.getcwd()
+    sha = None
+    diffs = []
+    branch = None
+    try:
+        os.chdir(os.path.dirname(pyomo.__file__))
+        sha = os.popen('git rev-parse HEAD').read().strip()
+        diffs = os.popen('git diff-index --name-only HEAD').read()
+        diffs = diffs.strip().split()
+        branch = os.popen('git symbolic-ref -q --short HEAD').read().strip()
+    finally:
+        os.chdir(cwd)
+    return { 'branch':branch, 'sha':sha, 'diffs':diffs,
+             'pyomo_version': pyomo_version }
+
+
+def getRunInfo(cython):
+    info = {
+        'time': time.time(),
+        'python_implementation': platform.python_implementation(),
+        'python_version': tuple(sys.version_info),
+        'python_build': platform.python_build(),
+        'platform': platform.system(),
+        'hostname': platform.node(),
+    }
+    if info['python_implementation'].lower() == 'pypy':
+        info['pypy_version'] = tuple(sys.pypy_version_info)
+    if cython:
+        import Cython
+        info['cython'] = tuple(int(x) for x in Cython.__version__.split('.'))
+    info.update(getPyomoInfo())
+    return info
+
+
+def run_tests(cython, argv):
+    gc.collect()
+    gc.collect()
+    results = ( getRunInfo(cython), OrderedDict() )
+    recorder = DataRecorder(results[1])
+    unittest.pytest.main(argv, plugins=[recorder])
+    gc.collect()
+    gc.collect()
+    return results
+
+
+def main(argv):
+    parser = argparse.ArgumentParser(
+        epilog="Remaining arguments are passed to nosetests"
```

Comment:
Is "nosetests" still correct here?

Reply:
Oh definitely not. Not correct at all.

Reply:
It is definitely not right. Should be pytest.

Reply:
fixed.